### ISSUE TITLE
Why doesn't ambiguous interface member lookup warn?

### ISSUE THREAD

**Issue by SLaks** [dotnet/roslyn#3784]
Source: 

```C#
interface IList {
    int Count { get; set; }
}

interface ICounter {
    void Count(int i);
}

interface IListCounter : IList, ICounter {}

void M(int count) {} 
void M(System.Action<int> func) {} 

IListCounter x;
M(x.Count);
```

In pre-Roslyn, this should give [CS0467 Ambiguity between method 'method' and non-method 'non-method'. Using method group.](https://msdn.microsoft.com/en-us/library/ms228509.aspx).

In Roslyn, it silently picks the method group.  Why was this warning removed?

**Comment by SLaks** [comment]
http://source.roslyn.io/#Microsoft.CodeAnalysis.CSharp/Errors/ErrorCode.cs,319

**Comment by jaredpar** [comment]
@AlekseyTs, @ljw1004 is this a result of the better betterness work?

**Comment by AlekseyTs** [comment]
No, Better Betterness doesn't affect name lookup.

**Comment by bbarry** [comment]
was commented in https://github.com/dotnet/roslyn/commit/ed31ebb07401be6357a27792fe5e3660a6cd7db9

**Comment by SLaks** [comment]
@bbarry That's just when they commented out the unused warning; that's not when the behavior was removed.

**Comment by bbarry** [comment]
yep diff says that, the line is unchanged otherwise since before the initial import

**Comment by jaredpar** [comment]
CC @gafter in case he knows whether or not this was intentional.  

Whether or not this was intentional, I don't think we'll be adding it back at this point.  Removing it was not a breaking change but adding it back definitely is.

**Comment by gafter** [comment]
I think we intentionally duplicated the name lookup for compat; I think it was not intentional to drop the warning. Opening this up to add to "strict" more.

**Comment by alrz** [comment]
I think this is related to https://github.com/dotnet/roslyn/issues/20605
